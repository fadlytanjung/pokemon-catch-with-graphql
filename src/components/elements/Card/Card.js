import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './styles.scoped.css';

export default function Card(props) {
  const { children, className, disabled, onClick, style, to, variant } = props;
  const customClass = [styles[variant], className].filter(Boolean).join(' ');

  if (to && /^https?:\/\//.test(to)) {
    return (<a className={customClass} disabled={disabled} href={to} style={style}>{children}</a>);
  }

  if (to) {
    return (<Link className={customClass} disabled={disabled} onClick={onClick}
      style={style} to={to}>{children}</Link>);
  }

  if (onClick) {
    return (<button className={customClass} disabled={disabled}
      onClick={onClick} style={style}>{children}</button>);
  }

  return <div className={customClass} disabled={disabled} style={style}>{children}</div>;
}

Card.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  onClick: null,
  style: undefined,
  to: '',
  variant: 'bordered',
};

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  style: PropTypes.object,
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  variant: PropTypes.oneOf(['bordered', 'hover']),
};
