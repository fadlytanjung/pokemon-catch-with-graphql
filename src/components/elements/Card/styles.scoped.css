.bordered,
.hover {
  background-color: #ffffff;
  border-radius: 0.5rem;
  display: block;
  font-size: 1rem;
  padding: 1rem;
  text-align: left;
  width: 100%;
}

.bordered {
  border: 0.0625rem solid #e5e6e6;
}

.hover {
  border: none;
  box-shadow: 0 0.25rem 0.625rem rgba(0, 0, 0, 0.2);
}

button.bordered,
button.hover {
  cursor: pointer;
}
