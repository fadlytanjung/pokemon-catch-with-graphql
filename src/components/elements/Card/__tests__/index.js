import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from '../Card';

describe('src/components/elements/Card', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Card />);
    expect(tree).toMatchSnapshot();
  });

  test('render external link', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Card to="https://www.indihome.co.id/" />);
    expect(tree).toMatchSnapshot();
  });

  test('render link', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Card to="/" />);
    expect(tree).toMatchSnapshot();
  });

  test('render button', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Card onClick={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
