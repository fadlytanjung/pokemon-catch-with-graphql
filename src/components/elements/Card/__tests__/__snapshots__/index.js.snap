// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Card render 1`] = `
<div
  className=""
  disabled={false}
/>
`;

exports[`src/components/elements/Card render button 1`] = `
<button
  className=""
  disabled={false}
  onClick={[MockFunction]}
/>
`;

exports[`src/components/elements/Card render external link 1`] = `
<a
  className=""
  disabled={false}
  href="https://www.indihome.co.id/"
/>
`;

exports[`src/components/elements/Card render link 1`] = `
<Link
  className=""
  disabled={false}
  onClick={null}
  to="/"
/>
`;
